CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -737,10 +737,17 @@ end
 --!param amount (number) This amount is added to the existing value for the attribute,
 --  and is then capped to be between 0 and 1.
 function Humanoid:changeAttribute(attribute, amount)
-  -- Receptionist is always 100% happy
-  if self.humanoid_class and self.humanoid_class == "Receptionist" and attribute == "happiness" then
-    self.attributes[attribute] = 1
-    return true
+  -- Handle some happiness special cases
+  if attribute == "happiness" and self.humanoid_class then
+    if self.humanoid_class == "Receptionist" then
+      -- A receptionist is never unhappy
+      self.attributes[attribute] = 1
+      return true
+    elseif self.profile and self.profile.wage >= self.max_salary then
+      -- A maximum salaried staff member is never unhappy
+      self.attributes[attribute] = 1
+      return true
+    end
```

Comment:
I'm happy to squash this down to a single `if` should that be desired. Though I didn't want to make the two "special cases" more unreadable by doing so.

Reply:
You changed it in the good direction. You don't want to squeeze code if it makes understanding more difficult, eg here you would merge two independent reasons for being ultimately happy.

Having each case handled separately is much simpler to understand.